controllers/certificaterequest/issue_certificate.go

```diff
@@ -210,3 +218,23 @@ func (r *CertificateRequestReconciler) IssueCertificate(reqLogger logr.Logger, c
 
 	return nil
 }
+
+func (r *CertificateRequestReconciler) FindZoneIDForChallenge(namespace string) (dnsZoneID string, err error) {
+	dnsZones := hivev1.DNSZoneList{}
+	err = r.Client.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: namespace})
+	if err != nil {
+		return "", err
+	}
+
+	if len(dnsZones.Items) != 1 {
+		return "", fmt.Errorf("%d dnsZone objects in a specific namespace found, expected 1 dnsZone", len(dnsZones.Items))
+	}
+
+	if *dnsZones.Items[0].Status.AWS.ZoneID != "" {
+		return filepath.Base(*dnsZones.Items[0].Status.AWS.ZoneID), nil
+	} else if *dnsZones.Items[0].Status.GCP.ZoneName != "" {
+		return *dnsZones.Items[0].Status.GCP.ZoneName, nil
+	} else {
+		return "", nil
+	}
```

Comment:
```suggestion
	} 
	
	return "", nil

```

Reply:
Should this return an error if no Zones are found?

Reply:
Yes, you could be right that we can return an error but, I wasn't sure how to perform check ZoneID for the Azure so, the else statement was for Azure.

Reply:
Does certman-operator support Azure? If so, then you'll need to implement the proper check. If not, then you can return the error and ignore Azure checks.

Reply:
https://github.com/openshift/hive/blob/master/apis/hive/v1/dnszone_types.go#L149-L150 <<<< Seems like AzureDNSZoneStatus struct is empty so, We don't need Azure condition.